src/grader/test_lib.ml

```diff
@@ -358,421 +400,439 @@ module Make
-                | ((Asttypes.Labelled n | Asttypes.Optional n), v) -> (n, v))
-              args in
-          add @@ on_function_call (fn, args) ;
-          default_mapper.expr mapper e
-      | { pexp_desc = Pexp_ident ident; _ } as e ->
-          treat_variable ident ;
-          default_mapper.expr mapper e
-      | { pexp_desc = Pexp_construct (ident, _); _ } as e ->
-          treat_module_prefixes ident ;
-          default_mapper.expr mapper e
-      | { pexp_desc = Pexp_record (fields, _); _ } as e ->
-          List.iter (fun (ident, _) -> treat_module_prefixes ident) fields ;
-          default_mapper.expr mapper e
-      | { pexp_desc = Pexp_field (_, ident); _ } as e ->
-          treat_module_prefixes ident ;
-          default_mapper.expr mapper e
-      | { pexp_desc = Pexp_setfield (_, ident, _); _ } as e ->
-          treat_module_prefixes ident ;
-          default_mapper.expr mapper e
-      | { pexp_desc = Pexp_new ident; _ } as e ->
-          treat_module_prefixes ident ;
-          default_mapper.expr mapper e
-      | { pexp_desc = Pexp_let (Asttypes.Nonrecursive, pvs, iexpr); _ } ->
-          let before = !variables in
-          let modules_before = !modules in
-          List.iter (fun { pvb_expr; _ } ->
-              ignore (mapper.expr mapper pvb_expr))
-            pvs ;
-          List.iter (fun { pvb_pat; _ } ->
-              ignore (mapper.pat mapper pvb_pat))
-            pvs ;
-          ignore (mapper.expr mapper iexpr) ;
-          variables := before ;
-          modules := modules_before ;
-          expr
-      | { pexp_desc = Pexp_let (Asttypes.Recursive, pvs, iexpr); _ } ->
-          let before = !variables in
-          let modules_before = !modules in
-          List.iter (fun { pvb_pat; _ } ->
-              ignore (mapper.pat mapper pvb_pat))
-            pvs ;
-          List.iter (fun { pvb_expr; _ } ->
-              ignore (mapper.expr mapper pvb_expr))
-            pvs ;
-          ignore (mapper.expr mapper iexpr) ;
-          variables := before ;
-          modules := modules_before ;
-          expr
-      | { pexp_desc = Pexp_for (pat, sexpr, eexpr, _, iexpr); _ } ->
-          let before = !variables in
-          let modules_before = !modules in
-          ignore (mapper.expr mapper sexpr) ;
-          ignore (mapper.expr mapper eexpr) ;
-          ignore (mapper.pat mapper pat) ;
-          ignore (mapper.expr mapper iexpr) ;
-          variables := before ;
-          modules := modules_before ;
-          expr
-      | { pexp_desc = Pexp_fun (label, vexpr, pat, iexpr); _ } ->
-          let before = !variables in
-          let modules_before = !modules in
-          (match vexpr with
-           | Some vexpr -> ignore (mapper.expr mapper vexpr)
-           | None -> ()) ;
-          (match label with
-           | Asttypes.Nolabel -> ()
-           | Asttypes.Labelled label
-           | Asttypes.Optional label ->
+           match List.rev txt with
```

Comment:
There are some trailing whitespaces at the beginning of each line. Please use 

http://batsov.com/articles/2011/11/25/emacs-tip-number-3-whitespace-cleanup/

to clean up your files before commit.